ring.go

```diff
@@ -0,0 +1,237 @@
+package redis
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/golang/groupcache/consistenthash"
```

Comment:
Any change you could copy and paste this across? Looks like it's short enough to be embedded. I prefer not to have external dependencies, especially when it comes to hashing.